code/modules/shuttle/navigation_computer.dm

```diff
@@ -147,12 +149,55 @@
 		var/list/to_remove = list()
 		to_remove += the_eye.placement_images
 		to_remove += the_eye.placed_images
+		to_remove += the_eye.extra_images
 		if(!see_hidden)
 			to_remove += SSshuttle.hidden_shuttle_turf_images
 
 		user.client.images -= to_remove
 		user.client.view_size.resetToDefault()
 
+/obj/machinery/computer/camera_advanced/shuttle_docker/proc/shuttle_turf_from_coords(list/coords)
+	var/mob/eye/camera/remote/shuttle_docker/the_eye = eyeobj
+	var/shuttleDir = shuttle_port.dir
+	var/curDir = the_eye.dir
+	var/list/adjustedCoords = coords.Copy()
+
+	// Rotate coords so  they match the current shuttle docking port's dir
+	if(turn(curDir, -90) == shuttleDir)
+		adjustedCoords[1] = coords[2] + y_offset
+		adjustedCoords[2] = -(coords[1] + x_offset)
+	else if(turn(curDir, 90) == shuttleDir)
+		adjustedCoords[1] = -(coords[2] + y_offset)
+		adjustedCoords[2] = coords[1] + x_offset
+	else if(turn(curDir, 180) == shuttleDir)
+		adjustedCoords[1] = -(coords[1] + x_offset)
+		adjustedCoords[2] = -(coords[2] + y_offset)
+	else
+		adjustedCoords[1] = coords[1] + x_offset
+		adjustedCoords[2] = coords[2] + y_offset
+
+	return locate(shuttle_port.x + adjustedCoords[1], shuttle_port.y + adjustedCoords[2], shuttle_port.z)
+
+/obj/machinery/computer/camera_advanced/shuttle_docker/proc/gatherNavComputerIcons()
+	var/mob/eye/camera/remote/shuttle_docker/the_eye = eyeobj
+	var/list/placement_image_cache = the_eye.placement_images
+	var/list/extra_image_cache = the_eye.extra_images
+	for(var/i in 1 to placement_image_cache.len)
+		var/image/placement_image = placement_image_cache[i]
+		var/list/coords = placement_image_cache[placement_image]
+		var/turf/shuttle_turf = shuttle_turf_from_coords(coords)
+		var/list/images_to_add = list()
+		for(var/atom/A in shuttle_turf)
+			SEND_SIGNAL(A, COMSIG_SHUTTLE_NAV_COMPUTER_IMAGE_REQUESTED, images_to_add)
```

Comment:
Also, to be honest, this signal could be sent to the turf, or better, the area, while the element should handle (un)registering the signal when the object it's attached to changes area or location, but I understand that's a few steps more complex. I personally have had some issues doing it for the beauty element for items, so this is more of a bonus.

Reply:
That was the original plan with this but I couldn't get it to work consistently.